android/src/main/java/com/pdftron/reactnative/views/DocumentView.java

```diff
@@ -1698,6 +1703,77 @@ public boolean handleBackButton() {
         return false;
     }
 
+    public void setPropertyForAnnotation(String annotId, int pageNumber, ReadableMap propertyMap) throws PDFNetException {
+        PDFViewCtrl pdfViewCtrl = getPdfViewCtrl();
+
+        boolean shouldUnlock = false;
+        try {
+            pdfViewCtrl.docLock(true);
+            shouldUnlock = true;
+
+            Annot annot = ViewerUtils.getAnnotById(pdfViewCtrl, annotId, pageNumber);
```

Comment:
pdfviewctrl.update(annot) is required at the end to update the location, you can try it out yourself, current code, after you call this method, annot is still in the old location...